src/Kestrel.Core/KestrelServerOptions.cs

```diff
@@ -100,8 +100,7 @@ public void Listen(IPEndPoint endPoint, Action<ListenOptions> configure)
                 throw new ArgumentNullException(nameof(configure));
             }
 
-            var listenOptions = new ListenOptions(endPoint) { KestrelServerOptions = this };
-            configure(listenOptions);
+            var listenOptions = new ListenOptions(endPoint) { KestrelServerOptions = this, Configure = configure };
```

Comment:
I need to run this later so that I can wrap the connection limit around the entire pipeline.